CDCgov/prime-reportstream
# HL7 OBR Mapping for OML messages

This PR adds mapping to HL7 `OBR` fields for the `OML` message type. It also extracts the common field definitions for `ORU` and `OML` into a common file to be reused (`metadata/hl7_mapping/common/observation-request.yml`)

Test Steps:
1. Run `./gradlew primeCLI --args='fhirdata --output-format HL7 --input-file "./src/testIntegration/resources/datatests/FHIR_to_HL7/sample_OML_20230831-0001.fhir"  -s "./metadata/hl7_mapping/receivers/Flexion/TILabOrder.yml"'`
2. Check that the output HL7 message has the `OBR` field. Subfields to look for are `OBR-1`, `OBR-2`, `OBR-4` and `OBR-7`, `OBR-11`

## Changes
- In `prime-router/metadata/hl7_mapping/OML_O21/base/service-request-order-observation.yml`, added mapping for:
  - `OBR-1`: Set ID – OBR
  - `OBR-2`: Placer Order Number
  - `OBR-7`: Observation Date/Time
- Updated `prime-router/metadata/hl7_mapping/OML_O21/OML_O21-base.yml` with hl7 output 
- Extracted common field definitions in `ORU` and `OML` to `metadata/hl7_mapping/common/observation-request.yml`

## Checklist

### Testing

## Linked Issues
[Create Outbound OML OBR Segment](https://github.com/CDCgov/trusted-intermediary/issues/493)
